packages/core/src/lib/assets/createImagesContext.ts

```diff
@@ -0,0 +1,132 @@
+import path from 'path';
+import { v4 as uuid } from 'uuid';
+import fs from 'fs-extra';
+import fromBuffer from 'image-type';
+import imageSize from 'image-size';
+import { KeystoneConfig, ImageMetadata, ImagesContext } from '../../types';
+import { AssetsAPI } from './types';
+const DEFAULT_BASE_URL = '/images';
+export const DEFAULT_IMAGES_STORAGE_PATH = './public/images';
+
+export function getImageMetadataFromBuffer(buffer: Buffer): ImageMetadata {
+  const fileType = fromBuffer(buffer);
+  if (!fileType) {
+    throw new Error('File type not found');
+  }
+
+  const extension = fileType.ext;
+  if (extension !== 'jpg' && extension !== 'png' && extension !== 'webp' && extension !== 'gif') {
+    throw new Error(`${extension} is not a supported image type`);
+  }
+
+  const { height, width } = imageSize(buffer);
+
+  if (width === undefined || height === undefined) {
+    throw new Error('Height and width could not be found for image');
+  }
+  return { width, height, filesize: buffer.length, extension };
+}
+
+export function createImagesContext(
+  config: KeystoneConfig,
+  s3Assets: () => Map<string, AssetsAPI>
+): ImagesContext | undefined {
+  if (!config.storage) {
+    return;
+  }
```

Comment:
I kinda feel like we should just always have the image and file stuff on the context rather than now if there's 1 or more configs now that there's N configs rather than 1 or 0 like before, thoughts?